tests/ttnn/unit_tests/operations/test_moreh_dot.py

```diff
@@ -163,99 +105,33 @@ def test_moreh_matmul_1d(input_shape, dtype, device):
 @pytest.mark.parametrize(
     "dtype",
     (
-        ttnn.float32,
         ttnn.bfloat16,
         ttnn.int32,
         None,
     ),
 )
 def test_moreh_matmul_1d_callback(input_shape, dtype, device, use_program_cache):
     torch.manual_seed(3072)
-    # get tensors
-    output_shape = [1, 1, 1, 1]
-    if dtype is None:
-        npu_dtype = torch.bfloat16
-    npu_dtype = dtype
-    cpu_dtype = get_torch_dtype(dtype)
 
-    # tt matmul
-    cpu_layout = ttnn.ROW_MAJOR_LAYOUT
     for i in range(2):
-        tt_input, tt_other, _, _, _, _, torch_input, torch_other, _ = get_tensors(
-            input_shape, input_shape, output_shape, False, False, True, device, npu_dtype, cpu_dtype
-        )
-        tt_out = (
-            ttnn.operations.moreh.dot(tt_input, tt_other, dtype=dtype)
-            .cpu()
-            .to(cpu_layout)
-            .unpad_from_tile(output_shape)
-            .to_torch()
-        )
+        run_moreh_dot_test(input_shape, dtype, device)
         torch_dummy = torch.randn([32, 32])
-        tt_dummy = to_ttnn(torch_dummy, device=device)
+        tt_dummy = ttnn.from_torch(torch_dummy, device=device)
         if i == 0:
             num_program_cache_entries = device.num_program_cache_entries()
             assert num_program_cache_entries > 0
         else:
             assert device.num_program_cache_entries() == num_program_cache_entries
 
-    # torch matmul
-    torch_input = torch.reshape(torch_input, (torch_input.shape[-1],))
-    torch_other = torch.reshape(torch_other, (torch_other.shape[-1],))
-    torch_out = torch.matmul(torch_input, torch_other)
-
-    # test for equivalance
-    rtol = atol = 0.1
-    passing, output_pcc = comp_allclose_and_pcc(torch_out, tt_out[0][0][0][0], pcc=0.999, rtol=rtol, atol=atol)
-    logger.debug(f"Out passing={passing}")
-    logger.debug(f"Output pcc={output_pcc}")
-
-    assert passing
-
 
 @pytest.mark.parametrize(
     "input_shape",
     ([1, 1, 1, 10],),  # test not mutiple of 32 case
 )
 @pytest.mark.parametrize(
     "dtype",
-    (
-        ttnn.float32,
-        ttnn.bfloat16,
-        ttnn.int32,
-        None,
-    ),
+    (ttnn.bfloat16,),
 )
 def test_moreh_dot_optional_output_tensor(input_shape, dtype, device):
```

Comment:
Can you parametrize use_optional_output instead of making another test?

Reply:
Wouldn't that double the tests in one module? (Since we have to have 2 options: use optional output and not use)